src/palace/manager/api/axis.py

```diff
@@ -776,13 +781,28 @@ def catch_up_from(
         for bibliographic, circulation in self.api.recent_activity(start):
             self.process_book(bibliographic, circulation)
             count += 1
-            if count % self.batch_size == 0:
-                self._db.commit()
         progress.achievements = "Modified titles: %d." % count
 
+    @retry(
+        retry=(
+            retry_if_exception_type(StaleDataError)
+            | retry_if_exception_type(ObjectDeletedError)
+        ),
+        stop=stop_after_attempt(MAX_RETRIES),
+        wait=wait_exponential(multiplier=1, min=1, max=60),
+        reraise=True,
+    )
     def process_book(
         self, bibliographic: Metadata, circulation: CirculationData
     ) -> tuple[Edition, LicensePool]:
+        if self._db.dirty:
```

Comment:
I don't love this check here, but I couldn't think of a cleaner way to do it and still use `tenacity` which is a nice library.  Any suggestions here would be appreciated.

Reply:
I don't think this catches all the conditions you want to catch here... dirty just means that changes haven't been flushed, not that there are changes within the transaction. 

Does it make more sense to take a similar approach to overdrive and not bring in the `tenacity` library here?